addons/website/models/ir_http.py

```diff
@@ -438,9 +438,12 @@ def _xmlid_to_obj(cls, env, xmlid):
     @api.model
     def get_frontend_session_info(self):
         session_info = super(Http, self).get_frontend_session_info()
+        country_code = request.session.get('geoip', {}).get('country_code')
+        phone_code = request.env['res.country'].search([('code', '=', country_code)], limit=1).phone_code if country_code else None
```

Comment:
Hm. That would add a sql query on every page, right? For the sake of adding a phone prefix in case we find a form in the page.. Not convinced.

We tried to minimize those queries, see 
```python
'website_company_id': request.website._get_cached('company_id'),
```
From Website override of `get_frontend_session_info()`

Maybe this should be done either:
- By RPC once page is loaded, as we already do (we perform a read on partner, maybe this should not just be a read but a read + fetch other data such as phone on geoip)
- By using a cached method as we do for other model.. Not sure it is worth just for a phone tho, maybe you can quickly check if other places would benefit from a cache on this model (to avoid a query also elsewhere), see `def _get_cached()` (@Gorash will introduce a new one on ir.module.module so I guess it could become a good practice)

Reply:
:+1: 
I put `_get_cached_values`, it should be the same name (`_get_cached` or `_get_cached_values`) for consistency and pay attention to what is cached.

Reply:
I changed the form prefill mechanism to a dedicated route which also obtains the phone code.
The phone code is now obtained from a cache on `res.country` (made similarly to what was done for `res.lang`).
I also kept the `id`, `code` and `name` in the cache to keep it similar to the one in `res.lang`, even though those are not used yet.

Reply:
@rdeodoo @Gorash Ready for review (not the owner, cannot use the request button - nor update the title)